### ISSUE TITLE
[Blocked] Epic: Improve PWA capabilities

### ISSUE THREAD

**Issue by annarhughes** [chaynHQ/bloom-frontend#1197]
### Overview

**Blocked by app router not being implemented yet (issue #702)**  If you're interested in working on this task in the future, follow the issue and you'll receive an update when it's unblocked and ready to be worked on.

This epic aims to add new Progressive Web App (PWA) capabilities to Bloom, and breaks down the tasks into sub issues, one for each new feature. Each sub issue should be completed asynchronously on a separate PR.

Read more about PWA and it's capabilities [here](https://www.freecodecamp.org/news/what-are-progressive-web-apps-pwa-guide/) and [here](https://dev.to/udoka033/progressive-web-apps-pwa-a-comprehensive-guide-57ii#lighthouse)

**Our current PWA implementation:**
- Basic PWA functionality is set up using [`next-pwa`](https://github.com/shadowwalker/next-pwa)
- The app can be downloaded/installed from the browser, as a browser app. This adds the Bloom app to the users homescreen.
- Push notifications and offline capabilities are not setup

### Action Items
Note these subissues have not been created yet due to this epic being blocked by design decisions and app router migration (issue #702). If you're interested in working on this task in the future, follow the issue and you'll receive an update when it's unblocked and ready to be worked on.

- [ ] Replace existing [`next-pwa`](https://github.com/shadowwalker/next-pwa) setup with nextjs built in PWA functionality for app router - see nextjs [guide](https://nextjs.org/docs/app/building-your-application/configuring/progressive-web-apps). Blocked by app router not being implemented yet (issue #702). Add only the setup and logic, skip all steps that add UI - these will be completed separately.
- [ ] Add subscribe/unsubscribe to push notifications setting to the `/account` page, using the example in the [guide](https://nextjs.org/docs/app/building-your-application/configuring/progressive-web-apps#2-implementing-web-push-notifications) adjusting the style to our design. Blocked by design.
- [ ] Add CTA to install the app/add to homepage. There is an example implementation of this in the guide above, we'll replace the UI and add a "Download app" button for all non-iOS devices. Blocked by design.

**Comment by bl00dymarie** [comment]
We're closing this epic/issue for now. If needed, we'll open an updated version in the future.
